solr/core/src/java/org/apache/solr/handler/admin/ZookeeperInfoHandler.java

```diff
@@ -358,96 +357,45 @@ public void command() {
-
-    if (params.get("addr") != null) {
-      throw new SolrException(ErrorCode.BAD_REQUEST, "Illegal parameter \"addr\"");
-    }
-
-    String detailS = params.get(PARAM_DETAIL);
-    boolean detail = detailS != null && detailS.equals("true");
-
-    String dumpS = params.get("dump");
-    boolean dump = dumpS != null && dumpS.equals("true");
-
-    int start = params.getInt("start", 0); // Note start ignored if rows not specified
-    int rows = params.getInt("rows", -1);
-
-    String filterType = params.get("filterType");
-    if (filterType != null) {
-      filterType = filterType.trim().toLowerCase(Locale.ROOT);
-      if (filterType.length() == 0) filterType = null;
+    if (params.get(PATH) == null) {
+      V2ApiUtils.squashIntoNamedList(rsp.getValues(), zookeeperAPI.getFiles());
+    } else {
+      V2ApiUtils.squashIntoNamedList(rsp.getValues(), zookeeperAPI.getFile());
     }
-    FilterType type = (filterType != null) ? FilterType.valueOf(filterType) : FilterType.none;
+  }
 
-    String filter = (type != FilterType.none) ? params.get("filter") : null;
-    if (filter != null) {
-      filter = filter.trim();
-      if (filter.length() == 0) filter = null;
-    }
+  @Override
+  public Boolean registerV2() {
+    return true;
+  }
 
-    ZKPrinter printer = new ZKPrinter(cores.getZkController());
-    printer.detail = detail;
-    printer.dump = dump;
-    boolean isGraphView = "graph".equals(params.get("view"));
-    // There is no znode /clusterstate.json (removed in Solr 9), but we do as if there's one and
-    // return collection listing. Need to change services.js if cleaning up here, collection list is
-    // used from Admin UI Cloud - Graph
-    boolean paginateCollections = (isGraphView && "/clusterstate.json".equals(path));
-    printer.page = paginateCollections ? new PageOfCollections(start, rows, type, filter) : null;
-    printer.pagingSupport = pagingSupport;
-
-    try {
-      if (paginateCollections) {
-        // List collections and allow pagination, but no specific znode info like when looking at a
-        // normal ZK path
-        printer.printPaginatedCollections();
-      } else {
-        printer.print(path);
-      }
-    } finally {
-      printer.close();
-    }
-    rsp.getValues().add(RawResponseWriter.CONTENT, printer);
+  @Override
+  public Collection<Class<? extends JerseyResource>> getJerseyResources() {
+    return List.of(ZookeeperAPI.class);
   }
 
   // --------------------------------------------------------------------------------------
   //
   // --------------------------------------------------------------------------------------
 
-  static class ZKPrinter implements ContentStream {
+  public static class ZKPrinter implements ContentStream {
     static boolean FULLPATH_DEFAULT = false;
 
     boolean indent = true;
     boolean fullpath = FULLPATH_DEFAULT;
-    boolean detail = false;
-    boolean dump = false;
+    public boolean detail = false;
```

Comment:
[-0] I wonder about all the visibility changes in the nested-classes in this PR.  As I read this diff, they're not really used in this Handler any more since the logic has (rightly) all moved into the API class...could we avoid the visibility escalation by moving the classes out of here and into the one place they're actually used (i.e. `ZooKeeperAPI`)?

Alternatively, we could avoid making the internal fields public and instead add getters/setters for those fields.  Though that may not be worth the effort....